### PR TITLE
rgw: Use get_val to get configs values

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@shinobu-x I'm not sure when it will be safe to change all of these operations at once;  if we do, we'll need a round of changes to optimize at least some of the accesses (e.g., caching).

**Comment by mattbenjamin** [comment]
@adamemerson @cbodley casey was mentioning something like gperf could be helpful here?

**Comment by adamemerson** [comment]
@mattbenjamin @cbodley If we wanted to get rid of the std::map and do this sort of thing at compile time we could use gperf and just turn all our value lookups into array lookups.

**Comment by shinobu-x** [comment]
test this please

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -6232,7 +6232,7 @@ int main(int argc, const char **argv)
     RGWMetadataLog *meta_log = store->meta_mgr->get_log(period_id);
 
     formatter->open_array_section("entries");
-    for (; i < g_ceph_context->_conf->rgw_md_log_max_shards; i++) {
+    for (; i < g_ceph_context->_conf->get_val<int64_t>("rgw_md_log_max_shards"); i++) {
```
Comment (shinobu-x) [inline src/rgw/rgw_admin.cc]: I was going to make local variable for `rgw_md_log_max_shards` for instance.
`g_ceph_context->_conf->get_val<int64_t>("rgw_md_log_max_shards")` should not be called in same method several time.

Comment (mattbenjamin) [reply src/rgw/rgw_admin.cc]: @shinobu-x sure, if you can find any instance;  I think we should hook up the caching api (and in core look at perfect hashing &c), but those can be separate changes;  it seems clear we need to move forward

**Pull request closed without merging**